src/components/FilterEditor.tsx

```diff
@@ -155,17 +156,17 @@ export default class FilterEditor extends React.Component<FilterEditorProps, Fil
     })
   }
 
-  static getDerivedStateFromProps(props: FilterEditorProps, currentState: FilterEditorState) {
+  public static getDerivedStateFromProps: GetDerivedStateFromProps<FilterEditorProps & WithTranslation, FilterEditorState> = (props, state) => {
```

Comment:
Why was this converted to an arrow function?

Reply:
Ah, sorry, I was getting a type error when passing the class to `withTranslation`. I think I can fix it another way. I think it was complaining that the props weren't `Readonly<T>`, so I just defaulted to importing the `GetDerivedStateFromProps` type from React. I'll double check.